src/web/components/SharingPermission/SharingPermissionsTable.tsx

```diff
@@ -3,41 +3,55 @@ import { CheckedState } from '@radix-ui/react-checkbox';
 import { useEffect, useMemo, useState } from 'react';
 
 import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
-import { AvailableParticipantDTO } from '../../../api/routers/participantsRouter';
+import { useAvailableSiteList } from '../../services/site';
 import { Dialog } from '../Core/Dialog';
+import { Loading } from '../Core/Loading';
 import { MultiSelectDropdown } from '../Core/MultiSelectDropdown';
 import { SortableTableHeader } from '../Core/SortableTableHeader';
 import { TriStateCheckbox, TriStateCheckboxState } from '../Core/TriStateCheckbox';
-import { ParticipantsTable } from './ParticipantsTable';
+import { ParticipantsTable, SharingParticipant } from './ParticipantsTable';
 
 import './SharingPermissionsTable.scss';
 
 type SharingPermissionsTableProps = {
-  sharingParticipants: AvailableParticipantDTO[];
+  sharedSiteIds: number[];
+  sharedTypes: string[];
   onDeleteSharingPermission: (siteIds: number[]) => Promise<void>;
   participantTypes: ParticipantTypeDTO[];
 };
 
-function NoParticipant() {
-  return (
-    <div className='no-participants-container'>
-      <img src='/group-icon.svg' alt='group-icon' />
-      <div className='no-participants-text'>
-        <h2>No Participants</h2>
-        <span>You don&apos;t have any sharing permissions yet.</span>
-      </div>
-    </div>
-  );
-}
-
 export function SharingPermissionsTable({
-  sharingParticipants,
+  sharedSiteIds,
+  sharedTypes,
   onDeleteSharingPermission,
   participantTypes,
 }: SharingPermissionsTableProps) {
+  const { sites: availableParticipants, isLoading } = useAvailableSiteList();
   const [filterText, setFilterText] = useState('');
   const [checkedParticipants, setCheckedParticipants] = useState<Set<number>>(new Set());
   const [openConfirmation, setOpenConfirmation] = useState(false);
+  const siteIds = useMemo(() => new Set(sharedSiteIds), [sharedSiteIds]);
```

Comment:
What's the benefit of using this rather than the `sharedSiteIds` prop directly? I see you utilise some of the Set methods (e.g. `.has`) - does this also provide a performance benefit?

Reply:
Yes, I found we had a few place that we need to do `siteIds.includes` which is O(n), and `.has` is O(1)

Reply:
good to know! ta

Reply:
`useMemo` is probably unnecessary here - it has an overhead, and the cost of the operation (creating a set from an array) is trivial compared to the cost of rendering the component, so avoiding the set creation every time the component is re-rendered is unnecessary.

You should use `useMemo` if the operation is expensive, or if keeping the same reference will let you avoid re-rendering another expensive component (i.e. you pass the memoized value as a prop to another component which is expensive to re-render).

Reply:
There's a good article about when to *not* use `useMemo` here: https://blog.logrocket.com/when-not-to-use-usememo-react-hook/

Reply:
This siteIds is used in the other `useMemo` hook, which is used to populate the addedBy field to the participant and get the sharingParticipants list. This siteIds without the useMemo hook will cause re-render of that hook.

I've noticed we have lots of redundant renders on this page, I would like to create a ticket for it and look into that once we have some spare time in the sprint. Hopefully we could clean up the usage of unnecessary hooks, is that okay?

Reply:
I'm trying to work out why the other `useMemo` exists - what was the actual problem?